rinex/src/lib.rs

```diff
@@ -2905,19 +2852,88 @@ impl Dcb for Rinex {
+//        /*
+//         * Determine mean value of all observed Phase and Pseudo Range
+//         * observations, for all Sv
+//         */
+//        let mut mean: HashMap<Sv, HashMap<Observable, f64>> = HashMap::new();
+//        /*
+//         * Associate a Phase code to all PR codes
+//         */
+//        let mut associations: HashMap<Observable, Observable> = HashMap::new();
+//
+//        let pr_codes: Vec<Observable> = self.observable()
+//            .filter_map(|obs|
+//                if obs.is_pseudorange_observable() {
+//                    Some(obs.clone())
+//                } else {
+//                    None
+//                })
+//            .collect();
+//
+//        for observable in self.observable() {
+//            if !observable.is_phase_observable() {
+//                if !observable.is_pseudorange_observable() {
+//                    continue;
+//                }
+//            }
+//            // code associations (for future combianations)
+//            if observable.is_phase_observable() {
+//                for pr_code in &pr_codes {
+//
+//                }
+//            }
+//
+//            for sv in self.sv() {
+//                /*
+//                 * average phase values
+//                 */
+//                let phases: Vec<f64> = self.carrier_phase()
+//                    .filter_map(|(_, svnn, obs, ph)| {
+//                        if sv == svnn && observable == obs {
+//                            Some(ph)
+//                        } else {
+//                            None
+//                        }
+//                    })
+//                    .collect();
+//                if let Some(data) = mean.get_mut(&sv) {
+//                    data.insert(observable.clone(), phases.mean());
+//                } else {
+//                    let mut map: HashMap<Observable, f64> = HashMap::new();
+//                    map.insert(observable.clone(), phases.mean());
+//                    mean.insert(sv, map);
+//                }
+//                /*
+//                 * average PR values
+//                 */
+//                let pr: Vec<f64> = self.pseudo_range()
+//                    .filter_map(|(_, svnn, obs, pr)| {
+//                        if sv == svnn && observable == obs {
+//                            Some(pr)
+//                        } else {
+//                            None
+//                        }
+//                    })
+//                    .collect();
+//                if let Some(data) = mean.get_mut(&sv) {
+//                    data.insert(observable.clone(), pr.mean());
+//                } else {
+//                    let mut map: HashMap<Observable, f64> = HashMap::new();
+//                    map.insert(observable.clone(), pr.mean());
+//                    mean.insert(sv, map);
+//                }
+//            }
+//        }
+//        HashMap::new()
+//    }
+//}
```

Comment:
`Mp()` will needs the average Phase/pseudo range,  that can easily be evluated with `statrs`.
This equation has never been completely evaluated, and MP is a complex topic. 
Will work on this in the future